x/mint/keeper/keeper.go

```diff
@@ -101,14 +106,65 @@ func (k Keeper) MintCoins(ctx context.Context, newCoins sdk.Coins) error {
 	return k.bankKeeper.MintCoins(ctx, types.ModuleName, newCoins)
 }
 
-// AddCollectedFees implements an alias call to the underlying supply keeper's
-// AddCollectedFees to be used in BeginBlocker.
-func (k Keeper) AddCollectedFees(ctx context.Context, fees sdk.Coins) error {
-	return k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, k.feeCollectorName, fees)
+// MoveCoinsFromMintToEcosystem moves freshly minted tokens from the mint module
+// which has permissions to create new tokens, to the ecosystem account which
+// only has permissions to hold tokens.
+func (k Keeper) MoveCoinsFromMintToEcosystem(ctx context.Context, mintedCoins sdk.Coins) error {
+	fmt.Println("MoveCoinsFromMintToEcosystem, moving: ", mintedCoins)
+	if mintedCoins.Empty() {
+		return nil
+	}
+	return k.bankKeeper.SendCoinsFromModuleToModule(
+		ctx,
+		types.ModuleName,
+		types.EcosystemModuleName,
+		mintedCoins,
+	)
+}
+
+// PayCosmosValidatorRewardFromEcosystemAccount sends funds from the ecosystem
+// treasury account to the validator rewards account (fee collector)
+// PayCosmosValidatorRewardFromEcosystemAccount to be used in BeginBlocker.
+func (k Keeper) PayCosmosValidatorRewardFromEcosystemAccount(ctx context.Context, rewards sdk.Coins) error {
+	fmt.Println("PayCosmosValidatorRewardFromEcosystemAccount, paying: ", rewards)
```

Comment:
we dont actually want print statements right, can we remove them all?

Reply:
yes good catch